File: src/modifiers/contrast.ts
import { createModifier } from './createModifier';

/// SwiftUI `.contrast(_:)`. Multiplies the view's color contrast by
/// `value`. `1` is unchanged, `0` is solid gray, values `> 1` increase
/// contrast, and negative values invert colors.
export function contrast(value: number) {
  return createModifier('contrast', { value });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/Contrast.swift
import SwiftUI

/// SwiftUI `.contrast(_:)`. Multiplies color contrast by `value`;
/// `1` leaves the view unchanged.
enum RNWContrastModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("contrast") { view, params, _ in
            AnyView(view.contrast(params.double("value") ?? 1))
        }
    }
}
